src/functor.jl

```diff
@@ -42,3 +55,25 @@ function fmap(f, x; cache = IdDict())
   haskey(cache, x) && return cache[x]
   cache[x] = isleaf(x) ? f(x) : fmap1(x -> fmap(f, x, cache = cache), x)
 end
+
+"""
+    fcollect(x; exclude = v -> false)
+
+Traverse `x` by recursing each child of `x` as defined by [`functor`](@ref)
+and collecting the results into a flat array.
+
+Doesn't recurse inside branches rooted at nodes `v`
+for which `exclude(v) == true`.
+In such cases, the root `v` is also excluded from the result.
+By default, `exclude` always yields `false`. 
+
+See also [`children`](@ref).
+"""
+function fcollect(x; cache = [], exclude = v -> false)
+  x in cache && return cache
+  if !exclude(x)
```

Comment:
How do you determine what to exclude? Are you suggesting tagging individual arrays somehow?

Reply:
It's a user provided argument. By default it doesn't exclude anything.

Reply:
As I understand it, you can't exclude specific arrays, so maybe we need an MWE on how it's expected to work?

Reply:
> As I understand it, you can't exclude specific arrays

It would be up to the user to figure out some tagging to do this. The main use-case isn't this, but the interface allows for it if that's what the user wants.

Reply:
if you want to exclude a branch rooted at some specific array `a`, you do `exclude = v -> v === a`. If you want to exclude based on some types, you do `exclude = v -> v isa SomeTypeUnion`.

Reply:
> There are some examples in the OP. @CarloLucibello should we put those into the docstring?

yes, I can add some examples to the docstring